fhir-core/src/main/java/com/ibm/fhir/core/util/ResourceTypeHelper.java

```diff
@@ -19,90 +19,208 @@
-                .filter(rtn -> !R4B_ONLY_RESOURCE_TYPES.contains(rtn))
-                .filter(rtn -> !ABSTRACT_TYPES.contains(rtn))
+    private static final Set<String> R4_COMPATIBLE_RESOURCES = Collections.unmodifiableSet(new LinkedHashSet<>(
+            R4_ENUMS.stream()
+                .filter(rtn -> !ABSTRACT_TYPE_ENUMS.contains(rtn))
                 .map(ResourceTypeName::value)
                 .collect(Collectors.toList())));
 
-    private static final Set<String> R4B_RESOURCES = Collections.unmodifiableSet(new LinkedHashSet<>(
-            Arrays.stream(ResourceTypeName.values())
-                .filter(rtn -> !REMOVED_RESOURCE_TYPES.contains(rtn))
-                .filter(rtn -> !ABSTRACT_TYPES.contains(rtn))
+    private static final Set<String> R4B_COMPATIBLE_RESOURCES = Collections.unmodifiableSet(new LinkedHashSet<>(
+            R4B_ENUMS.stream()
+                .filter(rtn -> !ABSTRACT_TYPE_ENUMS.contains(rtn))
                 .map(ResourceTypeName::value)
                 .collect(Collectors.toList())));
 
-    private static final Set<String> R4B_ONLY_RESOURCES = Collections.unmodifiableSet(new LinkedHashSet<>(
-            R4B_ONLY_RESOURCE_TYPES.stream()
+    private static final Set<String> R4_AND_R4B_COMPATIBLE_RESOURCES = Collections.unmodifiableSet(new LinkedHashSet<>(
+            R4B_ENUMS.stream()
+                .filter(rtn -> !R4B_ONLY_RESOURCE_ENUMS.contains(rtn))
+                .filter(rtn -> !BACKWARD_BREAKING_R4B_ENUMS.contains(rtn))
+                .filter(rtn -> !ABSTRACT_TYPE_ENUMS.contains(rtn))
                 .map(ResourceTypeName::value)
                 .collect(Collectors.toList())));
 
     private static final Set<String> ABSTRACT_RESOURCES = Collections.unmodifiableSet(
-            ABSTRACT_TYPES.stream()
+            ABSTRACT_TYPE_ENUMS.stream()
+                .map(ResourceTypeName::value)
+                .collect(Collectors.toSet()));
+
+    private static final Set<String> BACKWARD_BREAKING_R4B_RESOURCES = Collections.unmodifiableSet(
+            BACKWARD_BREAKING_R4B_ENUMS.stream()
+                .map(ResourceTypeName::value)
+                .collect(Collectors.toSet()));
+
+    private static final Set<String> FORWARD_BREAKING_R4B_RESOURCES = Collections.unmodifiableSet(
+            FORWARD_BREAKING_R4B_ENUMS.stream()
                 .map(ResourceTypeName::value)
                 .collect(Collectors.toSet()));
 
     /**
      * @param fhirVersion The value of the MIME-type parameter 'fhirVersion' for the current interaction
-     *          (e.g. "4.3" not "4.3.0")
-     * @return a set of resource type names that corresponds to the requested fhirVersion
+     * @return the set of resource type names from FHIR R4B that are backwards-compatible with the requested fhirVersion
      */
-    public static Set<String> getResourceTypesFor(FHIRVersionParam fhirVersion) {
+    public static Set<String> getR4bResourceTypesFor(FHIRVersionParam fhirVersion) {
```

Comment:
on the fence about whether to keep this flavor for convenience or force everything into `getCompatibleResourceTypes(sourceFhirVersion, targetFhirVersion)`.  thoughts?